### PR TITLE
add impl && unit tests

### PR DESCRIPTION
**Pull request by weilueluo**
https://jira-eng-rtp3.cisco.com/jira/browse/LH-71074

### CODE REVIEW

File: client/model/device/status/status.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package status
+
+type Type string
+
+// see also: https://github.com/cisco-lockhart/lh-plugin-core/blob/master/lh-target/src/main/java/com/cisco/lockhart/target/data/ServiceActivationState.java
```
Comment (weilueluo) [inline client/model/device/status/status.go:5]: hmm remove this link?

File: client/connector/connectoronboarding/create_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+				)
+			},
+			assertFunc: func(output *connectoronboarding.CreateOutput, err error, t *testing.T) {
+				assert.NotNil(t, output)
+				assert.Nil(t, err)
+				assert.Equal(t, *output, activeConnector)
+			},
+		},
+		{
+			testName: "should ends on connector Active status after an Onboarding state",
```
Comment (siddhuwarrier) [inline client/connector/connectoronboarding/create_test.go:67]: would reword this test case: should retry until SDC activation status is `ACTIVE`

File: client/connector/connectoronboarding/retry.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+)
+
+func UntilConnectorStatusIsActive(ctx context.Context, client http.Client, readInp connector.ReadByNameInput, readOutp *connector.ReadOutput) retry.Func {
+	return func() (bool, error) {
+		readRes, err := connector.ReadByName(ctx, client, readInp)
+		*readOutp = *readRes
+		if err != nil {
+			return false, err
+		}
+		client.Logger.Printf("connector larStatus: %v\n", readRes.ConnectorStatus)
```
Comment (siddhuwarrier) [inline client/connector/connectoronboarding/retry.go:18]: Nit: I wouldn't use the term `larStatus` as you've made a bunch of effort to hide `lar` here :)

File: provider/internal/connector/connectoronboarding/resource.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	Name types.String `tfsdk:"name"`
+}
+
+func (r *Resource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_sdc_onboarding"
+}
+
+func (r *Resource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Provides an SDC connector resource. This allows SDC to be onboarded, updated, and deleted on CDO.",
```
Comment (siddhuwarrier) [inline provider/internal/connector/connectoronboarding/resource.go:39]: Rephrase: `Use this resource to wait for an SDC to finish onboarding. When an SDC is onboarded, either manually or using the CDO Terraform Modules for [AWS](linkhere) and [vSphere](linkhere), it can take a few minutes before the SDC is active and capable of proxying communications between CDO and the device. This resource allows you to wait until this is done.`

File: provider/internal/connector/connectoronboarding/resource.go [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "TODO",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "TODO",
```
Comment (siddhuwarrier) [inline provider/internal/connector/connectoronboarding/resource.go:50]: Rephrase: `Specify the name of the SDC.`

**Review by siddhuwarrier — CHANGES_REQUESTED** [review]

File: client/connector/connectoronboarding/create_test.go (hunk 2 above)
Comment (weilueluo) [reply client/connector/connectoronboarding/create_test.go:67]: done 👍

File: client/connector/connectoronboarding/retry.go (hunk 3 above)
Comment (weilueluo) [reply client/connector/connectoronboarding/retry.go:18]: yea, done

File: provider/internal/connector/connectoronboarding/resource.go (hunk 5 above)
Comment (weilueluo) [reply provider/internal/connector/connectoronboarding/resource.go:50]: done 👍

File: provider/internal/connector/connectoronboarding/resource.go (hunk 4 above)
Comment (weilueluo) [reply provider/internal/connector/connectoronboarding/resource.go]: done 👍

**Review by siddhuwarrier — APPROVED** [review]
Fix terraform-aws-cdo-sdc link before merging. Thanks
